openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java

```diff
@@ -0,0 +1,386 @@
+            for (IntObjectCursor<IntHashSet> superSuperCell : superSuperCells) {
+                IntHashSet newSuperCell = new IntHashSet();
+                for (IntCursor cell : superSuperCell.value)
+                    newSuperCell.addAll(superCells.get(cell.value));
+                superSuperCellsToBaseCells.put(superSuperCell.key, newSuperCell);
+            }
+            superSuperCellsToBaseCells.putAll(superCells);
+            superCells.putAll(superSuperCells);
+
+            //Calculate the concave hull for all super(super)cells
+            for (IntObjectCursor<IntHashSet> superCell : superSuperCellsToBaseCells) {
+
+                List<Coordinate> superCellCoordinates = createSuperCellCoordinates(superCell.value);
+
+                LineString ring = createContour(superCellCoordinates);
+                if (ring == null || ring.getNumPoints() < 2) {
+                    cellStorage.setCellContourOrder(superCell.key, new ArrayList<>(), new ArrayList<>());
+                    continue;
+                }
+
+                expandAndSaveContour(superCell.key, ring);
+            }
+        }
+        return superCells;
+    }
+
+    private List<Coordinate> createSuperCellCoordinates(IntHashSet superCell) {
+        List<Coordinate> superCellCoordinates = new ArrayList<>(superCell.size() * 10);
+        for (IntCursor subcell : superCell) {
+            List<Double> subCellContour = cellStorage.getCellContourOrder(subcell.value);
+            int j = 0;
+            while (j < subCellContour.size()) {
+                double lat = subCellContour.get(j);
+                j++;
+                double lon = subCellContour.get(j);
+                j++;
+                superCellCoordinates.add(new Coordinate(lon, lat));
+            }
+        }
+        //Need to sort the coordinates, because they will be added to a search tree
+        //The order of insertion changes the search tree coordinates and we want consistency between runs
+        Collections.sort(superCellCoordinates);
+        return superCellCoordinates;
+    }
+
+    private Geometry concHullOfNodes(List<Coordinate> coordinates) {
+        int j = 0;
+        double defaultVisitorThreshold = 0.0035;
+        double defaultSearchWidth = 0.0008;
+        double defaulPointWidth = 0.005;
+
+        List<Coordinate> points = new ArrayList<>(1 / 20 * coordinates.size());
+        PointItemVisitor visitor = new PointItemVisitor(0, 0, defaultVisitorThreshold);
+        Quadtree qtree = new Quadtree();
+        Envelope searchEnv = new Envelope();
+        TreeSet<Coordinate> treeSet = new TreeSet<>();
+
+        while (j < coordinates.size()) {
+            double latitude = coordinates.get(j).y;
+            double longitude = coordinates.get(j).x;
+            j++;
+            addPoint(visitor, points, qtree, searchEnv, treeSet, longitude, latitude, defaultSearchWidth, defaulPointWidth, true);
```

Comment:
Because this whole method is from the original Isochrones code. I have no use for the return value, but I am reusing code without going into the details of how it works. Refactoring the old isochrones code would give an opportunity to reevaluate the reason why there is a return value in the first place. Refactoring of the old isochrones code is not part of this project, though.